include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -416,6 +416,50 @@ __get_accessor_ptr(const _Acc& __acc)
 #endif
 }
 
+class __event
+{
+    sycl::event __m_event;
+    sycl::queue __m_queue;
+    bool __in_order = false;
+
+  public:
+    __event(sycl::event __e): __m_event(__e) {}
+    __event() {}
+
+    __event(sycl::queue __q): __m_queue(__q), __in_order(true) {}
+
+    void wait()
+    {
+        if(__in_order)
+            __m_queue.wait_and_throw();
+        else
+            __m_event.wait_and_throw();
+    }
+
+    bool has_event() const { return !__in_order; }
```

Comment:
Is this a problem if the default constructor was called to create this `__event`?  `__in_order` is false, but only a default constructed event exists.  

I suppose in the way it is used currently, this isnt a problem.  However, the naming of this function may be misleading.  Maybe we would just be better off to `assert(!__in_order);` directly in the conversion function.

Reply:
Actually, I've redesigned that code a little bit.